documentation/release_notes.markdown

```diff
@@ -4,116 +4,104 @@ title: "Puppet Server: Release Notes"
 
-This is a bug-fix release that resolves a disruptive logging configuration issue.
+For more information, see the following [Trapperkeeper project](https://tickets.puppetlabs.com/browse/TK) tickets:
 
-### Bug fix: Puppet Server starts when configured to log to syslog
+-   [TK-211: Trapperkeeper doesn't error if two services implementing the same protocol are started](https://tickets.puppetlabs.com/browse/TK-211)
+-   [TK-347: Support directories and paths in TK's "bootstrap-config" CLI argument](https://tickets.puppetlabs.com/browse/TK-347)
+-   [TK-349: TK should not fail during startup if an unrecognized service is found in bootstrap config](https://tickets.puppetlabs.com/browse/TK-349)
+-   [TK-351: Ensure all bootstrap related errors log what file they come from](https://tickets.puppetlabs.com/browse/TK-351)
 
-If its Logback service is configured to log to syslog, Puppet Server 2.3.0 fails to start. Puppet Server 2.3.1 fixes this regression, which did not affect prior versions of Puppet Server.
-
-* [SERVER-1215](https://tickets.puppet.com/browse/SERVER-1215)
-
-### All changes
+Also, see these related tickets:
 
-* [All Puppet Server issues targeted at this release](https://tickets.puppet.com/issues/?jql=project%20%3D%20SERVER%20AND%20fixVersion%20%3D%20%22SERVER%202.3.1%22%20ORDER%20BY%20updated%20DESC%2C%20priority%20DESC%2C%20created%20ASC)
+-   [EZ-72: Implement support for overriding bootstrap-config path in app's project.clj](https://tickets.puppetlabs.com/browse/EZ-72)
 
-## Puppet Server 2.3.0
+### Improvement: Responses to unauthenticated HTTPS requests include less information
 
-Released March 16, 2016.
+When responding to unauthorized HTTPS requests, previous versions of Puppet Server 2.x returned the requester's IP address and [authorization rule][auth.conf] in addition to logging the failed request. Puppet Server 2.4.0 removes this information from the response and directs the responder to consult the server logs for details.
 
-This is a feature release that adds functionality for static catalogs, a new environment classes API, and restarting Puppet Server with a HUP signal.
+-   [TK-360: Error messages returned to client should not include IP or rule blocking the request](https://tickets.puppetlabs.com/browse/TK-360)
 
-> ### New requirements
->
-> Puppet masters running Puppet Server 2.3 depend on [Puppet Agent 1.4.0](https://docs.puppet.com/puppet/4.4/reference/about_agent.html) or newer, which installs [Puppet 4.4](https://docs.puppet.com/puppet/4.4/) and compatible versions of its related tools and dependencies on the server. Puppet agents running older versions of Puppet Agent can connect to Puppet Server 2.3 --- this requirement applies to the Puppet Agent running on the Puppet Server node *only*.
+### New feature: X.509-compliant certificate extensions can match authorization rules
```

Comment:
I would probably move this up to the top of the list, or, put it just after the sections about Ubuntu/systemd, because this is definitely the most compelling thing that lands in this release other than possibly the Ubuntu support.